### DISCUSSION TITLE
Frames, slides and pages

### DISCUSSION THREAD

**Discussion by josephwright** [josephwright/ltx-talk#14] [General]
Currently, `beamer` uses the environment name `frame` at the document level, with each page within that referred to as a 'slide'. However, that means there's a clash with LaTeX's `\frame` command and some internal gymnastics. On one had, the naming is well-established, on the other we should consider if we want to carry this forward or to adjust (`slide` environment plus `overlay` or `page`?).

**Comment by wspr** [comment]
From my Australian context, calling it a "slide" would match the terminology I would use colloquially and professionally (e.g. "see the figure on the left of the slide"). Although overlays and so on create additional PDF pages, I don't really think about the discrete PDF pages conceptually as separate "slides". OTOH, I guess in the old days of overhead projectors, every additional transparency you printed was almost literally a physical "slide", so I think the terminology can be argued both ways.

I think I agree with the idea of `slide` + `overlay`, potentially with a compatibility switch to use the old syntax if desired.

I had never considered the clash between \frame and \begin{frame} — that must indeed be awkward internally :)

**Reply by FrankMittelbach** [reply]
I'm old enough to have used slides (still have many of those) and even have a printer which can generate them if need be :-). From those days I think the terminology that beamer currently use is correct. The frame environment generates one or more slides:
- a base one
- and a number of overlay slides

You put them on top of each other on an overhead projector (for those who do not know what that is, watch any of Don Knuth's videos :-) ) and to not do what Don often does you need to align them properly, which means put them all in the same frame (if you have one).

So botton line, I think the terminology is next to perfect (i.e., inside the environment all these overlays are in the same frame) the only catch is the clash with \frame.

To that I see two solutions
- allow \frame inside being reverted to the LaTeX definition
- or stop allowing \frame{....} to be a command form of the beamer environment meaning

I know that some people use `\frame{ \maketitle}` in their slide decks and such decks would break in xbeamer but ...

My vote would go for keeping the terminology and choose one or the other of the two solutions (with me leaning towards forbidding beamer \frame command form).

**Reply by josephwright** [reply]
Oh, definitely plan to drop the `\frame{\maketitle}` support - Till deprecated it a _long_ time ago there

**Reply by josephwright** [reply]
> So botton line, I think the terminology is next to perfect (i.e., inside the environment all these overlays are in the same frame) the only catch is the clash with \frame.

I think we are resolved here to keep the current naming from `beamer` [and I'm also old enough to have had lectures on 35 mm slides :)]

**Reply by FrankMittelbach** [reply]
> Oh, definitely plan to drop the `\frame{\maketitle}` support - Till deprecated it a _long_ time ago there

then all that is needed is to implement the env support that allows \frame and \begin{frame} to have different code paths :-)

**Reply by josephwright** [reply]
Oh, that's already done: `\frame` gives a 'LaTeX `frame`' when used inside `\begin{frame}`. (We need to think about non-frame text, but that's for later.)

**Comment by josephwright** [comment]
Supporting LaTeX `\frame` strictly inside the `frame` environment is not too bad, but I agree that it's probably better to deprecate. I think 'slide' is the term used generally - if you look at e.g. PowerPoint or Keynote, they describe things that way. (I'm old enough to have had lectures using actual 35 mm slides!)

What I think we can't do is not _support_ `frame`, at least other than with some 'strictly new syntax' option. But if the general feeling is `slide` is the name to promote, I'll adjust.

**Reply by FrankMittelbach** [reply]
Well, I'm not that much in favor of changing frame (env) to slide (env) as it  initially invalids each an every existing deck. While it is not a major problem to change via global replace all occurances of `{frame}` into `{slide}` it is nevertheless yet another hurdle especially if you just want to find out if your deck works with xbeamer without problems. Besides, as I said I think the terminology is actually more correct to talk about the slides as being the individual pages in the PDF. But it wouldn't be the end of the world for me if the decision goes the other way :-)

**Comment by josephwright** [comment]
[Some thumbs-up/down votes please :)]

**Comment by davidcarlisle** [comment]
`slide` looks good to me, beamer gets away with `\frame`/`begin{frame}` as the latex version is more or less undocumented, it's not clear why it's there really it's sort of `\fbox` but with zero sep and using the `picture` mode rule width. So a simpler, and syntactically cleaner, option than making both work is simply to over-write the latex definition, perhaps making it available as `\latexframe` or some such,  just in case.

**Reply by josephwright** [reply]
Looking back, I suspect this is the answer that works best - unless others object.

**Reply by Skillmon** [reply]
Hm, a simple `\let` at each `\begin{frame}` isn't expensive, I'm not entirely sold.

**Reply by josephwright** [reply]
@Skillmon Sure, it's not that which worries me - I'm just not that keen on the overloading of `\frame` (but then also see the parallel discussion about the fact that starting today I think we'd separate `\foo` from `\begin{foo}` entirely).

**Comment by samcarter** [comment]
> `slide` looks good to me, beamer gets away with `\frame`/`begin{frame}` as the latex version is more or less undocumented, it's not clear why it's there really it's sort of `\fbox` but with zero sep and using the `picture` mode rule width. So a simpler, and syntactically cleaner, option than making both work is simply to over-write the latex definition, perhaps making it available as `\latexframe` or some such, just in case.

In all the years of reading beamer questions, I've seen only one or two cases of people directly using the latex `\frame` within the frame (there might be more case where `\frame` is indirectly used by something else).

**Reply by FrankMittelbach** [reply]
it would blow up with any internal use of \frame and that is used in a few places though not that many across all of TL

**Comment by cfr42** [comment]
I'd keep `page` for the actual pages of the PDF, though. Right now it is possible explain how  beamer works by saying that every overlay creates another page and it is quite important to be able to explain that, I think.
